### DISCUSSION TITLE
Character limit in textarea field doesn't restrict characters

### DISCUSSION THREAD

**Discussion by jayrdi** [statamic/cms#8441] [Q&A]
I'm trying to restrict the number of chars a user can enter for an article summary to display on an index card.

I have the fieldtype "textarea" with the "character limit" set to 150.

If you go into the News Article it allows you to enter more than 150 chars though, and the char counter under the window just displays e.g. 167/150 in red.

Is there a way to actually prevent users typing more than 150 characters easily? I know that they will ignore this char counter and just enter whatever they like!

Thanks

**Comment by duncanmcclean** [comment]
I don't think so - it's a decent [feature request](https://github.com/statamic/ideas) though.

**Comment by jackmcdade** [comment]
It won’t let you save if you have too many. Just like with Twitter, we allow you to enter too many so have a chance to edit it down.

**Reply by jayrdi** [reply]
That's funny, I can save and it then displays the "too long" text on the site...

Is that because this is a local build?

**Reply by jackmcdade** [reply]
Can you show a screenshot?

**Reply by jayrdi** [reply]
@joshuablum gave me a solution by adding `max:150` under validation as a rule, and it now works as expected.

So, it's my interpretation of the fields that wrong rather than the functionality!

"Character limit" displays the char counter and `max:150` validation enforces it. Is that right?

Thanks for your help - loving Statamic 😁

**Reply by jackmcdade** [reply]
Ah yeah my bad too — that _is_ the right solution. Thanks @joshuablum! And glad you're enjoying it @jayrdi!
